inc/classes/subscriber/third-party/plugins/class-smush-subscriber.php

```diff
@@ -0,0 +1,76 @@
+<?php
+namespace WP_Rocket\Subscriber\Third_Party\Plugins;
+
+use WP_Rocket\Logger\Logger;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+/**
+ * Subscriber for compatibility with Smush
+ *
+ * @since  3.4.2
+ * @author Soponar Cristina
+ */
+class Smush_Subscriber implements Subscriber_Interface {
+	/**
+	 * Subscribed events for Smush.
+	 *
+	 * @since  3.4.2
+	 * @author Soponar Cristina
+	 * @inheritDoc
+	 */
+	public static function get_subscribed_events() {
+		if ( ! defined( 'WP_SMUSH_VERSION' ) ) {
+			return [];
+		}
+
+		return [
+			'update_option_wp-smush-settings'      => [ 'smush_maybe_deactivate_lazyload', 11 ],
+			'activate_wp-smushit/wp-smush.php'     => [ 'smush_maybe_deactivate_lazyload', 11 ],
+			'rocket_maybe_disable_lazyload_helper' => 'is_smush_lazyload_active',
+		];
+	}
+
+	/**
+	 * Disable WP Rocket lazyload when activating WP Smush and values are already in the database.
+	 *
+	 * @since  3.4.2
+	 * @author Soponar Cristina
+	 */
+	public function smush_maybe_deactivate_lazyload() {
+		$lazy_load_option = get_option( WP_SMUSH_PREFIX . 'settings' );
+		$lazyload         = isset( $lazy_load_option['lazy_load'] ) ? $lazy_load_option['lazy_load'] : false;
+
+		if ( ! $lazyload ) {
+			return;
+		}
+
+		update_rocket_option( 'lazyload', 0 );
+	}
+
+	/**
+	 * Disable WP Rocket lazyload fields if WP Smush lazyload is enabled
+	 *
+	 * @since  3.4.2
+	 * @author Soponar Cristina
+	 *
+	 * @return string
+	 */
+	public function is_smush_lazyload_active( $rocket_maybe_disable_lazyload_plugins ) {
+		if ( ! defined( 'WP_SMUSH_VERSION' ) ) {
+			return;
+		}
+
+		$lazy_load_option = get_option( WP_SMUSH_PREFIX . 'settings' );
+		$lazyload         = isset( $lazy_load_option['lazy_load'] ) ? $lazy_load_option['lazy_load'] : false;
+
+		if ( ! $lazyload ) {
+			return $rocket_maybe_disable_lazyload_plugins;
+		}
+
+		if ( is_plugin_active( 'wp-smushit/wp-smush.php' ) && ! empty( $lazyload ) ) {
```

Comment:
`&& ! empty( $lazyload )` is useless since it cannot be empty (tested line 66).
Also, seeing how this method is used, this `is_plugin_active( 'wp-smushit/wp-smush.php' )` test seems useless: if `WP_SMUSH_VERSION` is defined, then the plugin is active. It could be useful on a page where the plugin could have been disabled earlier (the constant would still be defined, but the plugin deactivated), is it the case here?

Reply:
Indeed is useless. Thank you for pointing this out. Removed it.